common/Reducers.jsm

```diff
@@ -465,7 +468,14 @@ function DiscoveryStream(prevState = INITIAL_STATE.DiscoveryStream, action) {
     case at.DISCOVERY_STREAM_LAYOUT_RESET:
       return {...prevState, lastUpdated: INITIAL_STATE.DiscoveryStream.lastUpdated, layout: INITIAL_STATE.DiscoveryStream.layout};
     case at.DISCOVERY_STREAM_FEEDS_UPDATE:
-      return {...prevState, feeds: action.data || prevState.feeds};
+      return {
+        ...prevState,
+        feeds: {
+          ...prevState.feeds,
```

Comment:
It's not obvious to me from reading this code why ...prevState.feeds is necessary.  If it is truly necessary, I think a comment in the code explaining why would be helpful.

Reply:
I'm just being careful that if other data ends up being placed on that in a future patch, I won't end up clobbering it.

It's not actually a requirement to my patch.